File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	total := 0

	for _, bird := range birdsPerDay {
		total += bird
	}
	return total
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	total := 0

	for day := 7 * (week - 1); day < 7*(week-1)+7; day++ {
		total += birdsPerDay[day]
	}
	return total
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for day := 0; day < len(birdsPerDay); day += 2 {
		birdsPerDay[day]++
	}
	return birdsPerDay
}

File: solutions/swift/bird-watcher/1/Sources/BirdWatcher/BirdWatcher.swift
func totalBirdCount(_ birdsPerDay: [Int]) -> Int { birdsPerDay.reduce(0, +) }

func birdsInWeek(_ birdsPerDay: [Int], weekNumber: Int) -> Int {
  birdsPerDay.dropFirst(7 * (weekNumber - 1)).prefix(7).reduce(0, +)
}

func fixBirdCountLog(_ birdsPerDay: [Int]) -> [Int] {
  birdsPerDay.enumerated().map { (index, bird) in bird + 1 - index % 2 }
}
